Plugin.php

```diff
@@ -52,43 +60,167 @@ public function boot()
+            Config::set('debugbar.collectors.route', false);
+
+            Event::listen('cms.page.beforeDisplay', function (CmsController $controller, $url, ?Page $page) use ($debugBar) {
+                if ($page) {
+                    $collector = new CmsCollector($controller, $url, $page);
+                    if (!$debugBar->hasCollector($collector->getName())) {
+                        $debugBar->addCollector($collector);
+                    }
+                }
+            });
+        }
+
+        if (Config::get('debugbar.collectors.components', true)) {
+            Event::listen('cms.page.initComponents', function (CmsController $controller, ?Page $page, ?Layout $layout) use ($debugBar) {
+                if ($page) {
+                    $collector = new ComponentsCollector($controller, $page, $layout);
+                    if (!$debugBar->hasCollector($collector->getName())) {
+                        $debugBar->addCollector($collector);
+                    }
+                }
+            });
+        }
+    }
+
+    /**
+     * Add collectors used by the Backend only
+     */
+    public function addBackendCollectors()
+    {
+        /** @var \Barryvdh\Debugbar\LaravelDebugbar $debugBar */
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+
+        if (Config::get('debugbar.collectors.backend', true)) {
+            Event::listen('backend.page.beforeDisplay', function (BackendController $controller, $action, array $params) use ($debugBar) {
+                $collector = new BackendCollector($controller, $action, $params);
+                if (!$debugBar->hasCollector($collector->getName())) {
+                    $debugBar->addCollector($collector);
+                }
+            });
+        }
+    }
+
+    /**
+     * Registers extensions in the CMS Twig environment
+     */
+    protected function registerCmsTwigExtensions()
+    {
+        $profile = new Profile;
+        $debugBar = $this->app->make(\Barryvdh\Debugbar\LaravelDebugbar::class);
+
+        Event::listen('cms.page.beforeDisplay', function ($controller, $url, $page) use ($profile, $debugBar) {
             $twig = $controller->getTwig();
             if (!$twig->hasExtension(\Barryvdh\Debugbar\Twig\Extension\Debug::class)) {
                 $twig->addExtension(new \Barryvdh\Debugbar\Twig\Extension\Debug($this->app));
                 $twig->addExtension(new \Barryvdh\Debugbar\Twig\Extension\Stopwatch($this->app));
             }
+
+            if (!$twig->hasExtension(ProfilerExtension::class)) {
+                $twig->addExtension(new ProfilerExtension($profile));
+            }
         });
+
+        if (class_exists(\DebugBar\Bridge\NamespacedTwigProfileCollector::class)) {
+            $debugBar->addCollector(new \DebugBar\Bridge\NamespacedTwigProfileCollector($profile));
+        } else {
+            $debugBar->addCollector(new \DebugBar\Bridge\TwigProfileCollector($profile));
+        }
```

Comment:
What causes that class to exist or not?

Reply:
It appears to be an incoming feature with the PHP Debug Bar, and brings in support for Twig 3's namespacing, which will be necessary if we roll out Twig 3 with the L9 update.